pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,390 @@
-			fakeClient.MockGet = func(ctx context.Context, key runtimeclient.ObjectKey, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
-				return fmt.Errorf("some error")
+	tests := map[string]struct {
+		namespace      string
+		name           string
+		name1          string
+		labelKey       string
+		labelValue     string
+		labelKey1      string
+		labelValue1    string
+		expectedMsg    string
+		labelSelector  string
+		expectedOutput string
+		lsKey          string
+		lsValue        string
+	}{
+		//operator and non-operator deployments
+		"OperatorAndNonOperatorHostDeployment": {
+			namespace:     "toolchain-host-operator",
+			name:          "host-operator-controller-manager",
+			name1:         "registration-service",
+			labelKey:      "kubesaw-control-plane",
+			labelValue:    "kubesaw-controller-manager",
+			labelKey1:     "toolchain.dev.openshift.com/provider",
+			labelValue1:   "codeready-toolchain",
+			expectedMsg:   "deployment \"host-operator-controller-manager\" successfully rolled out\n",
+			labelSelector: "kubesaw-control-plane=kubesaw-controller-manager",
+			lsKey:         "host",
+			lsValue:       "operator",
+		},
+		//operator and non-operator deployments, checking for autoscaler deployments,
+		//it should be treated as no non-operator deployment available
+		"OperatorAndNonOperatorWithAutoscalerDeployment": {
+			namespace:     "toolchain-member-operator",
+			name:          "member-operator-controller-manager",
+			name1:         "autoscaling-buffer",
+			labelKey:      "kubesaw-control-plane",
+			labelValue:    "kubesaw-controller-manager",
+			labelKey1:     "toolchain.dev.openshift.com/provider",
+			labelValue1:   "codeready-toolchain",
+			expectedMsg:   "deployment \"member-operator-controller-manager\" successfully rolled out\n",
+			labelSelector: "kubesaw-control-plane=kubesaw-controller-manager",
+			lsKey:         "host",
+			lsValue:       "operator",
+		},
+		//only non-operator deployment
+		"NonOperatorHostDeployment": {
+			namespace:      "toolchain-host-operator",
+			name:           "registration-service",
+			labelKey:       "toolchain.dev.openshift.com/provider",
+			labelValue:     "codeready-toolchain",
+			expectedMsg:    "deployment \"registration-service\" successfully rolled out\n",
+			labelSelector:  "toolchain.dev.openshift.com/provider=codeready-toolchain",
+			expectedOutput: "deployment.apps/registration-service restarted\n",
+		},
+		//only operator deployment
+		"OperatorHostDeployment": {
+			namespace:     "toolchain-host-operator",
+			name:          "host-operator-controller-manager",
+			labelKey:      "kubesaw-control-plane",
+			labelValue:    "kubesaw-controller-manager",
+			expectedMsg:   "deployment \"host-operator-controller-manager\" successfully rolled out\n",
+			labelSelector: "kubesaw-control-plane=kubesaw-controller-manager",
+			lsKey:         "host",
+			lsValue:       "operator",
+		},
+	}
```

Comment:
this whole setup and also the if-else flow in the test itself is a bit overcomplicated. 
The test execution via the combinations is useful for simple setup when it safes the code and simplifies the logic. This is not the case. You verify 4 test-cases, I believe that it would be much easier to read and understand if they were implemented as separate sub-tests. Let's not over-use this way of executing tests and let's not use it in cases when it doesn't make much sense.
You can still extract some of the common setup code to separate functions to minimize copy-paste

Reply:
yes right, updated it